OneSignalSDK/onesignal/src/main/java/com/onesignal/OSInAppMessageController.java

```diff
@@ -409,6 +429,62 @@ else if (action.getUrlTarget() == OSInAppMessageAction.OSInAppMessageActionUrlTy
         }
     }
 
+    private void saveViewedPageIdsToPrefs() {
+        OneSignalPrefs.saveStringSet(
+                OneSignalPrefs.PREFS_ONESIGNAL,
+                OneSignalPrefs.PREFS_OS_PAGE_IMPRESSIONED_IAMS,
+                // Post success, store impressioned pages to disk
+                viewedPageIds);
+    }
+
+    private void fireRESTCallForPageChange(@NonNull final OSInAppMessage message, @NonNull final OSInAppMessagePage page) {
+        final String variantId = variantIdForMessage(message);
+        if (variantId == null)
+            return;
+
+        final String pageId = page.getPageId();
+
+        final String messagePrefixedPageId = message.messageId + pageId;
+
+        // Never send multiple page impressions for the same message UUID unless that page change is from an IAM with redisplay
+        if (message.getViewedPageIds().contains(pageId) || viewedPageIds.contains(messagePrefixedPageId)) {
```

Comment:
same question as on iOS, see https://github.com/OneSignal/OneSignal-iOS-SDK/pull/735#discussion_r542662673

Reply:
same fix as ios. only tracking globally now, but it is still per inapp since we don't want them to affect each other (in local testing I was getting duplicate page ids for different inapps which is why I prefixed the page id with the message id)